src/query/service/src/interpreters/interpreter_insert.rs

```diff
@@ -297,22 +208,6 @@ impl InsertInterpreter {
             match may_error {
                 Some(error) => {
                     tracing::error!("insert stage file error: {}", error);
-                    GlobalIORuntime::instance()
-                        .block_on(async move {
-                            if stage_info.copy_options.purge {
-                                info!(
-                                    "insert: try to purge files:{}, elapsed:{}",
-                                    all_source_files.len(),
-                                    start.elapsed().as_secs()
-                                );
-                                Self::try_purge_files(ctx.clone(), &stage_info, &all_source_files)
```

Comment:
@everpcpc  please have a look, seems that commit 3f2a530d (fix: also purge files when failure) purges copied files on purpose.  thanks.

Reply:
We purge files on failure because we treat the 3-step insert as a whole process. When insert fails, we upload file to s3 again in previous design.

Reply:
I think it might be a redundant step, since we could rewrite the logic to reinsert/rereplace stage files as a whole, since the stage files would not be loaded if they commited in copyinto step